components/board/pi/impl/board.go

```diff
@@ -788,9 +788,17 @@ func pigpioInterruptCallback(gpio, level int, rawTick uint32) {
 		}
 		// this should *not* block for long otherwise the lock
 		// will be held
-		err := i.Tick(instance.cancelCtx, high, tick*1000)
-		if err != nil {
-			instance.logger.Error(err)
+		switch di := i.(type) {
+		case *BasicDigitalInterrupt:
+			err := Tick(instance.cancelCtx, di, high, tick*1000)
+			if err != nil {
+				instance.logger.Error(err)
+			}
+		case *ServoDigitalInterrupt:
+			err := ServoTick(instance.cancelCtx, di, high, tick*1000)
+			if err != nil {
+				instance.logger.Error(err)
+			}
 		}
```

Comment:
Put in a default case for anything unexpected. Otherwise, in the future there will be some third interrupt type and we're going to get confused why it has no errors but also doesn't work.